app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ReferCaseController.scala

```diff
@@ -70,13 +58,65 @@ class ReferCaseController @Inject()(verify: RequestActions,
       renderView(c => c.status == REFERRED, c => successful(views.html.confirm_refer_case(c)))
     }
 
-  def showContactInformation(reference: String): Action[AnyContent] =
-    (verify.authenticated
-      andThen verify.casePermissions(reference)
-      andThen verify.mustHave(Permission.REFER_CASE)).async { implicit request =>
-      renderView(c => c.status == OPEN, c => successful(views.html.refer_case_error(c)))
+  def postReferCase(reference: String): Action[MultipartFormData[Files.TemporaryFile]] = (verify.authenticated andThen verify.casePermissions(reference) andThen
+    verify.mustHave(Permission.REFER_CASE)).async(parse.multipartFormData) { implicit request: AuthenticatedCaseRequest[MultipartFormData[Files.TemporaryFile]] =>
+
+    val myForm = (checkReasonIsSelected andThen checkedOtherCommentNotEmpty) (ReferCaseForm.form.bindFromRequest())
+
+    def foldForm(error: String): Future[Result] = {
```

Comment:
I think this could be named a little better, also made`private`

Reply:
maybe `failWithFormError`

Reply:
Can it be made private given where it is? It's defined as a nested method so only has scope within it's enclosing method anyway

Reply:
Updated name though

Reply:
@ed0906  That was the assumption for those method declarations. Being inside the method the only method that can access them is the inner method. Therefore they will not be accessible from any other place (even more restrictive than private).